src/peer_digest.cc

```diff
@@ -705,26 +662,16 @@ peerDigestReqFinish(DigestFetchState * fetch, char * /* buf */,
             pd->times.retry_delay = 0;
             peerDigestSetCheck(pd, peerDigestNewDelay(fetch->entry));
         }
+        pd->finish(fetch, err);
```

Comment:
Please move the whole `if` statement code above inside PeerDigest::noteFetchFinished(). We will only check for pd validity in this function and then call noteFetchFinished() to do the rest.

Reply:
Done.